internal/providers/qemu/qemu_blockdev.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2020 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build s390x ppc64le
+
+// The QEMU provider on s390x and ppc64le fetch a configuration file from an
+// attached block device with serial/id 'ignition'.
+
+package qemu
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"time"
+
+	"github.com/coreos/ignition/v2/config/v3_1_experimental/types"
+	"github.com/coreos/ignition/v2/internal/log"
+	"github.com/coreos/ignition/v2/internal/providers/util"
+	"github.com/coreos/ignition/v2/internal/resource"
+
+	"github.com/coreos/vcontext/report"
+)
+
+const (
+	ignitionBlockDevicePath = "/dev/disk/by-id/virtio-ignition"
+)
+
+func FetchConfig(f *resource.Fetcher) (types.Config, report.Report, error) {
+	f.Logger.Warning("Fetching the Ignition config via the Virtio block driver is currently experimental and subject to change.")
+
+	_, err := f.Logger.LogCmd(exec.Command("modprobe", "virtio_blk"), "loading Virtio block driver module")
```

Comment:
I think a common pattern BTW is to do these in `ExecStartPre=` since `modprobe` is a privileged operation, but then after that one commonly wants to drop privileges.

Reply:
Yeah, mostly following the pattern of `qemu_fw_cfg` here.

Reply:
Going to mark this as resolved to make review easier, but we can split this out into a separate issue for further discussion.